src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1444,12 +1473,32 @@ String printCert(boolean isTsCert, String tab, Certificate c,
         }
 
         if (x509Cert != null) {
+            PublicKey key = x509Cert.getPublicKey();
+            String sigalg = x509Cert.getSigAlgName();
 
-            certStr.append("\n").append(tab).append("[");
-
+            // Process the certificate in the signer's cert chain to see if
+            // weak algorithms are used, and provide warnings as needed.
             if (trustedCerts.contains(x509Cert)) {
+                // If the cert is trusted, only check its key size, but not its
+                // signature algorithm. This is because warning should not be
+                // generated for SHA-1 roots which are not an issue.
+                certStr.append("\n").append(tab)
+                        .append("Signature algorithm: ")
+                        .append(sigalg)
+                        .append(rb.getString("COMMA"))
+                        .append(checkWeakKey(key));
+
+                certStr.append("\n").append(tab).append("[");
```

Comment:
It's a little strange to leave the other half of the bracket (`certStr.append("]");` on line 1568) outside the if-else block. Can you please move it inside? Of course you will have to duplicate them.

Reply:
I'd prefer to keep it as is instead of duplicating the code in several places inside if block and else block.